src/gui/plugins/visualize_contacts/VisualizeContacts.cc

```diff
@@ -0,0 +1,393 @@
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/MainWindow.hh>
+
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/World.hh"
+#include "ignition/gazebo/EntityComponentManager.hh"
+#include "ignition/gazebo/gui/GuiEvents.hh"
+#include "ignition/gazebo/rendering/RenderUtil.hh"
+#include "ignition/gazebo/components/Collision.hh"
+#include "ignition/gazebo/components/ContactSensor.hh"
+#include "ignition/gazebo/components/ContactSensorData.hh"
+
+#include "VisualizeContacts.hh"
+
+namespace ignition
+{
+namespace gazebo
+{
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE
+{
+  /// \brief Private data class for VisualizeContacts
+  class VisualizeContactsPrivate
+  {
+    /// \brief Creates ContactSensorData for Collision components without a
+    /// Contact Sensor by requesting the /enable_contact service
+    /// \param[in] Reference to the GUI Entity Component Manager
+    public: void CreateCollisionData(EntityComponentManager &_ecm);
+
+    /// \brief Transport node
+    public: transport::Node node;
+
+    /// \brief Current state of the checkbox
+    public: bool checkboxState{false};
+
+    /// \brief Previous state of the checkbox
+    public: bool checkboxPrevState{false};
+
+    /// \brief Message for visualizing contact positions
+    public: ignition::msgs::Marker positionMarkerMsg;
+
+    /// \brief Radius of the visualized contact sphere
+    public: double contactRadius{0.10};
+
+    /// \brief Message for visualizing contact forces
+    public: ignition::msgs::Marker forceMarkerMsg;
+
+    /// \brief Scale of the visualized force cylinder. By scale we mean how
+    /// many meters per newton the marker will be long
+    public: double forceScale{0.40};
+
+    /// \brief Update time of the markers in milliseconds
+    public: int64_t markerLifetime{200};
+
+    /// \brief Simulation time for the last markers update
+    public: std::chrono::steady_clock::duration lastMarkersUpdateTime{0};
+
+    /// \brief Mutex for variable mutated by the checkbox and spinboxes
+    /// callbacks.
+    /// The variables are: checkboxState, contactRadius, forceScale and
+    /// markerLifetime
+    public: std::mutex serviceMutex;
+
+    /// \brief Initialization flag
+    public: bool initialized{false};
+
+    /// \brief Name of the world
+    public: std::string worldName;
+  };
+}
+}
+}
+
+using namespace ignition;
+using namespace gazebo;
+
+/////////////////////////////////////////////////
+VisualizeContacts::VisualizeContacts()
+  : GuiSystem(), dataPtr(new VisualizeContactsPrivate)
+{
+}
+
+/////////////////////////////////////////////////
+VisualizeContacts::~VisualizeContacts() = default;
+
+/////////////////////////////////////////////////
+void VisualizeContacts::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Visualize contacts";
+
+  ignition::gui::App()->findChild<
+    ignition::gui::MainWindow *>()->installEventFilter(this);
```

Comment:
I think this isn't necessary, because there's no `eventFilter` function to receive the filtered events. This is only needed by plugins that receive events.